drivers/gpio/gpio_renesas_ra8.c

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_ra8_gpio
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/irq.h>
+#include <soc.h>
+
+#define PRCR_KEY   0xA500U
+#define VBATT_PORT 4
+
+static const gpio_pin_t vbatt_pins[] = {
+	2U,
+	3U,
+	4U,
+};
+
+#define RA8_INVALID_PORT_ADDR 0UL
+
+#define RA8_GPIO_PORT_ADDR(nodelabel) \
+	COND_CODE_1(DT_NODE_EXISTS(DT_NODELABEL(nodelabel)), \
+	(DT_REG_ADDR(DT_NODELABEL(nodelabel))), (RA8_INVALID_PORT_ADDR))
+
+struct gpio_ra8_config {
+	struct gpio_driver_config common;
+	uint8_t port_num;
+	R_PORT0_Type *port;
+};
+
+struct gpio_ra8_data {
+	struct gpio_driver_data common;
+};
+
+static int gpio_ra8_pin_configure(const struct device *dev, gpio_pin_t pin, gpio_flags_t flags)
+{
+	const struct gpio_ra8_config *config = dev->config;
+
+	if (((flags & GPIO_INPUT) != 0U) && ((flags & GPIO_OUTPUT) != 0U)) {
+		return -ENOTSUP;
+	}
+
+	if ((flags & GPIO_PULL_DOWN) != 0U) {
+		return -ENOTSUP;
+	}
+
+	if (config->port_num == VBATT_PORT) {
+		uint32_t clear = 0;
+
+		for (int i = 0; i < ARRAY_SIZE(vbatt_pins); i++) {
+			if (vbatt_pins[i] == pin) {
+				WRITE_BIT(clear, i, 1);
+			}
+		}
+
+		R_SYSTEM->PRCR = ((R_SYSTEM->PRCR | PRCR_KEY) | R_SYSTEM_PRCR_PRC1_Msk);
+
+		R_SYSTEM->VBTICTLR &= (uint8_t)~clear;
+
+		R_SYSTEM->PRCR = ((R_SYSTEM->PRCR | PRCR_KEY) & (uint16_t)~R_SYSTEM_PRCR_PRC1_Msk);
+	}
+
+	R_PMISC->PWPRS = 0;
+	R_PMISC->PWPRS = BIT(R_PMISC_PWPR_PFSWE_Pos);
+
+	uint32_t value = R_PFS->PORT[config->port_num].PIN[pin].PmnPFS;
+
+	/* Change mode to general IO mode and disable IRQ and Analog input */
+	WRITE_BIT(value, R_PFS_PORT_PIN_PmnPFS_PMR_Pos, 0);
+	WRITE_BIT(value, R_PFS_PORT_PIN_PmnPFS_ASEL_Pos, 0);
+	WRITE_BIT(value, R_PFS_PORT_PIN_PmnPFS_ISEL_Pos, 0);
+
+	if ((flags & GPIO_OUTPUT) != 0U) {
```

Comment:
Please handle all properties.
https://github.com/zephyrproject-rtos/zephyr/pull/72150/files#diff-e4af11f77a8bbf03f9175f29acf04a4f2adc7eed434ef6ecfe4470fe66037b88R84-R88

Reply:
We  handle for all properties. Please check. Thank you.